modules/google_sql_database_instance/main.tf

```diff
@@ -69,7 +69,7 @@ resource "google_sql_database_instance" "instance" {
     disk_autoresize             = var.settings_disk_autoresize
     disk_autoresize_limit       = var.settings_disk_autoresize_limit
     disk_type                   = var.settings_disk_type
-    disk_size                   = var.settings_disk_size
+    disk_size                   = var.settings_disk_autoresize ? null : var.settings_disk_size
```

Comment:
What should be starting value of disk? You should have the default value of disk as 10GB and ignore the changes to the attribute disk_size in the life_cycle hooks.

Reply:
Why would we ignore the disk size @sunilhonest ?

Reply:
`life_cycle` already ignores updates to `disk_size`. We don't need an initial value, it's [optional](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance) and will default to the minimum size (10GB).
> [disk_size](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#disk_size) - (Optional) The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased. The minimum value is 10GB.

 If you think there's use-case where we need a manual initial size bigger than 10GB, we can only ignore it for read-replicas.

Reply:
Updated to ignore`settings_disk_size` only the read replica to allow a custom initial size for the master instance.

Reply:
@jai We have `disk_autoresize` enabled and hence GCP increases the disk size. TF apply will try to reset it to value passed(ex: 10 GB in our case) and hence we should ignore it in the life_cycle.